### ISSUE TITLE
Lost almost all data when upgrading sync client

### ISSUE THREAD

**Issue by dylanmorroll** [laurent22/joplin#5293]
I'm running Joplin 1.7.11 on Debian and I just got a popup in the GUI asking me to upgrade the sync client. I clicked okay, it asked me to restart the app and I did so. It upgraded, asked me to restart again, I did so. It then started syncing and almost everything got wiped. All my notebooks went, I have about 20 notes left in 'All notes' and 200 in conflicts, just 2 tags left and then a warning came up in the bottom right saying fail-safe mode was activated as 98% of my data was to be deleted.

Luckily I had recently synced everything to my windows client so I booted into windows and enabled flight mode and made a back up of all my data. But when I turned the internet on it started trashing all my notes there as well.

Not sure what happened here but it doesn't seem right that the GUI should prompt for an update that isn't stable without warning to make a backup or something first!

Update: I disabled fail-safe mode and the rest of my local notes and tags were deleted, just the conflicts are left. It appears that upgrading the sync client deleted all data on the cloud, which then got synced to my local client.

## Environment
Joplin version: 1.7.11
Platform: Linux
OS specifics: Debian Buster, Linux 5.10.0-8-amd64 x86_64

## Steps to reproduce
Not sure what to put here as I'm not sure when the popup for upgrading the sync client appears.

## Describe what you expected to happen
I imagine the sync client should be updated without any data loss.

## Logfile
Log files trimmed to today:
[log-12-08.txt](https://github.com/laurent22/joplin/files/6974285/log-12-08.txt)

**Comment by laurent22** [comment]
There are quite a few SQL errors on your server so I'm wondering if it's related?

**Comment by dylanmorroll** [comment]
It could be, I'm using a hosted Nextcloud instance on Tab.Digital

**Comment by laurent22** [comment]
One version of Nextcloud had a bug where they would report a file is missing (error 404) when an SQL error happens, so that could be related.

It's not clear either why it suddenly asks you to upgrade the sync target if you haven't upgraded the client.

**Comment by dylanmorroll** [comment]
Tab.Digital is running Nextcloud 21.0.3.1 currently.

**Comment by laurent22** [comment]
This is a Nextcloud bug (error 500 means server error) so not much we can do on client side, but you could ask Tab.Digital if you have an account with them.

**Comment by dylanmorroll** [comment]
Hmm yeah I kept having issues so I just swapped to Dropbox and it seems to be working okay. Will close this issues for now unless I experience it again as it appears it might have been a server side Nextcloud thing.

However it might be worth adding logging in somewhere in case something like this happens again to give a bit more insight? I.e. to be able to discern between whether it's a local or server side bug.
